File: app/src/main/java/com/example/verve_calc/MainActivity.java
package com.example.verve_calc;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import net.pubnative.lite.sdk.HyBidAdRequest;
import net.pubnative.lite.sdk.views.HyBidAdView;
import net.pubnative.lite.sdk.HyBidInterstitialAd;
import net.pubnative.lite.sdk.HyBidInterstitialAdLoadCallback;
import net.pubnative.lite.sdk.HyBid;
import net.pubnative.lite.sdk.HyBidBannerAd;
import net.pubnative.lite.sdk.HyBidBannerAdView;
import net.pubnative.lite.sdk.HyBidAdSize;
import net.pubnative.lite.sdk.HyBidBannerAdLoadCallback;

public class MainActivity extends AppCompatActivity {

    private TextView textViewResult;
    private StringBuilder currentNumber = new StringBuilder();
    private String operator = "";
    private double result = 0.0;
    private HyBidAdView bannerAdView;
    private HyBidInterstitialAd mInterstitial;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        // Initialize HyBid SDK
        HyBid.initialize(this, "YOUR_HYBID_APP_TOKEN");
        // Find the banner ad view
        bannerAdView = findViewById(R.id.bannerAdView);

        // Load Banner Ad
        loadBannerAd();
        // Load Interstitial Ad
        loadInterstitialAd();

        textViewResult = findViewById(R.id.textViewResult);

        // Set click listeners for number buttons
        setNumberButtonClickListeners();

        // Set click listeners for operator buttons
        setOperatorButtonClickListeners();

        // Set click listener for equals button
        Button buttonEquals = findViewById(R.id.buttonEquals);
        buttonEquals.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onEqualClick();
                interstitialAd = new HyBidInterstitialAd(this,"", new HyBidInterstitialAd.Listener() {

                if (interstitialAd != null && interstitialAd.isReady()) {
                    interstitialAd.show();
                }
            }
        });

        // Set click listener for clear button
        Button buttonClear = findViewById(R.id.buttonClear);
        buttonClear.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onClearClick();
            }
        });

        // Set click listener for dot button
        Button buttonDot = findViewById(R.id.buttonDot);
        buttonDot.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onDotClick();
            }
        });
    }
    //load banner ad
    private void loadBannerAd() {
            bannerAdView.load("ZONE_ID", new PNAdView.Listener() {
                @Override
                public void onAdLoaded() {

                }

                @Override
                public void onAdLoadFailed(Throwable error) {

                }

                @Override
                public void onAdImpression() {

                }

                @Override
                public void onAdClick() {

                }
            });
    }
    //load interstitial ad
    private void loadInterstitialAd() {
            mInterstitial = new HyBidInterstitialAd(this,"ZONE_ID", new HyBidInterstitialAd.Listener() {
                @Override
                public void onInterstitialLoaded() {

                }

                @Override
                public void onInterstitialLoadFailed(Throwable error) {

                }

                @Override
                public void onInterstitialImpression() {

                }

                @Override
                public void onInterstitialDismissed() {

                }

                @Override
                public void onInterstitialClick() {

                }
            });
            mInterstitial.load();
    }

    private void setNumberButtonClickListeners() {
        int[] numberButtonIds = {R.id.button0, R.id.button1, R.id.button2, R.id.button3, R.id.button4,
                R.id.button5, R.id.button6, R.id.button7, R.id.button8, R.id.button9};

        for (int buttonId : numberButtonIds) {
            Button button = findViewById(buttonId);
            button.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    onNumberClick(button.getText().toString());
                }
            });
        }
    }

    private void setOperatorButtonClickListeners() {
        int[] operatorButtonIds = {R.id.buttonAdd, R.id.buttonSubtract, R.id.buttonMultiply, R.id.buttonDivide};

        for (int buttonId : operatorButtonIds) {
            Button button = findViewById(buttonId);
            button.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    onOperatorClick(button.getText().toString());
                }
            });
        }
    }

    private void onNumberClick(String number) {
        currentNumber.append(number);
        updateDisplay();
    }

    private void onOperatorClick(String op) {
        if (currentNumber.length() > 0) {
            operator = op;
            result = Double.parseDouble(currentNumber.toString());
            currentNumber.setLength(0);
        }
    }

    private void onEqualClick() {
        if (currentNumber.length() > 0 && !operator.isEmpty())

File: app/src/main/java/com/example/verve_calc/MainActivity.kt
package com.example.verve_calc

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import com.example.verve_calc.ui.theme.Verve_CalcTheme
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {

    private var currentInput = ""
    private var currentOperator = ""
    private var operand1 = ""
    private var operand2 = ""

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Set click listeners for number buttons
        button0.setOnClickListener { onNumberClick("0") }
        button1.setOnClickListener { onNumberClick("1") }
        button2.setOnClickListener { onNumberClick("2") }
        button3.setOnClickListener { onNumberClick("3") }
        button4.setOnClickListener { onNumberClick("4") }
        button5.setOnClickListener { onNumberClick("5") }
        button6.setOnClickListener { onNumberClick("6") }
        button7.setOnClickListener { onNumberClick("7") }
        button8.setOnClickListener { onNumberClick("8") }
        button9.setOnClickListener { onNumberClick("9") }

        // Set click listener for operator buttons
        buttonAdd.setOnClickListener { onOperatorClick("+") }
        buttonSubtract.setOnClickListener { onOperatorClick("-") }
        buttonMultiply.setOnClickListener { onOperatorClick("*") }
        buttonDivide.setOnClickListener { onOperatorClick("/") }

        // Set click listener for equals button
        buttonEquals.setOnClickListener { onEqualsClick() }

        // Set click listener for clear button
        buttonClear.setOnClickListener { onClearClick() }
    }

    private fun onNumberClick(number: String) {
        currentInput += number
        textViewInput.text = currentInput
    }

    private fun onOperatorClick(operator: String) {
        if (currentInput.isNotEmpty()) {
            operand1 = currentInput
            currentInput = ""
            currentOperator = operator
        }
    }

    private fun onEqualsClick() {
        if (currentInput.isNotEmpty() && operand1.isNotEmpty() && currentOperator.isNotEmpty()) {
            operand2 = currentInput

            val result = when (currentOperator) {
                "+" -> operand1.toDouble() + operand2.toDouble()
                "-" -> operand1.toDouble() - operand2.toDouble()
                "*" -> operand1.toDouble() * operand2.toDouble()
                "/" -> {
                    if (operand2.toDouble() != 0.0) {
                        operand1.toDouble() / operand2.toDouble()
                    } else {
                        "Error"
                    }
                }
                else -> "Error"
            }

            currentInput = result.toString()
            textViewInput.text = currentInput
            currentOperator = ""
            operand1 = ""
            operand2 = ""
        }
    }

    private fun onClearClick() {
        currentInput = ""
        currentOperator = ""
        operand1 = ""
        operand2 = ""
        textViewInput.text = "0"
    }
}